cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactStore.java

```diff
@@ -1052,6 +1059,36 @@ private Scan scanAppClasses(NamespaceId namespace) {
     return new Scan(startRow, Bytes.stopKeyForPrefix(startRow));
   }
 
+  /**
+   * Filters the plugins contained in the {@link ArtifactMeta} by using {@link #isAllowed(PluginClass)}
+   *
+   * @param artifactMeta the artifact meta
+   * @return the {@link ArtifactMeta} containing only the plugins which are not excluded
+   */
+  private ArtifactMeta filterExcludedPlugins(ArtifactMeta artifactMeta) {
+    ArtifactClasses classes = artifactMeta.getClasses();
+    Set<PluginClass> filteredPlugins = classes.getPlugins().stream()
+      .filter(this::isAllowed).collect(Collectors.toSet());
+    ArtifactClasses filteredClasses = ArtifactClasses.builder()
+      .addApps(classes.getApps())
+      .addPlugins(filteredPlugins)
+      .build();
+    return new ArtifactMeta(filteredClasses, artifactMeta.getUsableBy(), artifactMeta.getProperties());
+  }
+
+  /**
+   * Checks whether the plugin is excluded from being displayed/used and should be filtered out. The exclusion is
+   * determined by the requirements of the plugin and the configuration for
+   * {@link Constants#REQUIREMENTS_BLACKLIST}
+   *
+   * @param pluginClass the plugins class to check
+   * @return true if the plugin should not be excluded and is allowed else false
+   */
+  private boolean isAllowed(PluginClass pluginClass) {
+    // if a plugin has any requirement which is marked excluded in the config then the plugin should not be allowed
+    return Sets.intersection(pluginClass.getRequirements(), excludedRequirements).isEmpty();
```

Comment:
Use `return pluginClass.getRequirements().stream().anyMatch(excludedRequirements::contains);` instead as there is no need to create a new set.

Reply:
Done.